### PR TITLE
artifactor: don't crash if we can't get test idents

### PR DESCRIPTION
**Pull request by vrutkovs**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 1f81f74012cb9e5c824e8f88cc7a8ad672afd114:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by psav** [comment]
So, Can you give me a definitive test that hits this problem, because this is something that @RonnyPfannschmidt said shouldn't happen. If you look back in the history of the file, you'll see we originally had a workaround. @RonnyPfannschmidt  removed it because he thought it could cause other issues, but the result is that we are still hitting that edge case. We need to figure out why.

**Comment by vrutkovs** [comment]
This happened on run 35x0KEpu:
```
2016-02-24 09:20:40,014 [I] Running command `cd /var/www/miq/vmdb; bin/rails runner /tmp/load_conf.rb` (utils/ssh.py:138)
2016-02-24 09:20:49,299 [I] Invoking callback for signal [server_details_changed] (utils/signals.py:113)
2016-02-24 09:20:49,304 [I] Invoking callback for signal [server_config_changed] (utils/signals.py:113)
2016-02-24 09:20:49,861 [I] Setting up provider: rhos7-ga (utils/providers.py:298)
2016-02-24 09:20:51,989 [I] Checked out webdriver container grave_yonath (utils/browser.py:250)
2016-02-24 09:20:51,994 [I] webdriver command executor set to http://ibm-x3530m4-05.cfme.lab.eng.rdu2.redhat.com:4966/wd/hub (utils/browser.py:147)
2016-02-24 09:20:51,999 [I] tests can be viewed via vnc on display vnc://ibm-x3530m4-05.cfme.lab.eng.rdu2.redhat.com:66 (utils/browser.py:148)
2016-02-24 09:21:34,986 [I] Beginning to fill in form... (cfme/web_ui/__init__.py:1282)
2016-02-24 09:22:02,184 [I] Navigating to clouds_provider_new (cfme/fixtures/pytest_selenium.py:1068)
2016-02-24 09:22:14,615 [I] Identifying locator for region not found (cfme/web_ui/__init__.py:155)
2016-02-24 09:22:16,541 [E] Could not determine the reason for failing the navigation.  Reraising.  Exception: Message: can't access dead object
Build info: version: '2.45.0', revision: '5017cb8', time: '2015-02-26 23:59:50'
System info: host: '2e0a2ac171c9', ip: '172.17.196.121', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-229.1.2.el7.x86_64', java.version: '1.8.0_45'
Driver info: driver.version: unknown
Stacktrace:
    at <anonymous class>.bot.dom.isElement (file:///tmp/anonymous251398743202883260webdriver-profile/extensions/fxdriver@googlecode.com/components/driver-component.js:8083)
    at <anonymous class>.bot.dom.isTextual (file:///tmp/anonymous251398743202883260webdriver-profile/extensions/fxdriver@googlecode.com/components/driver-component.js:8159)
    at <anonymous class>.bot.dom.isEditable (file:///tmp/anonymous251398743202883260webdriver-profile/extensions/fxdriver@googlecode.com/components/driver-component.js:8168)
    at <anonymous class>.FirefoxDriver.prototype.mouseClick (file:///tmp/anonymous251398743202883260webdriver-profile/extensions/fxdriver@googlecode.com/components/driver-component.js:10672)
    at <anonymous class>.DelayedCommand.prototype.executeInternal_/h (file:///tmp/anonymous251398743202883260webdriver-profile/extensions/fxdriver@googlecode.com/components/command-processor.js:12274)
    at <anonymous class>.DelayedCommand.prototype.executeInternal_ (file:///tmp/anonymous251398743202883260webdriver-profile/extensions/fxdriver@googlecode.com/components/command-processor.js:12279)
    at <anonymous class>.DelayedCommand.prototype.execute/< (file:///tmp/anonymous251398743202883260webdriver-profile/extensions/fxdriver@googlecode.com/components/command-processor.js:12221) (cfme/fixtures/pytest_selenium.py:1152)
2016-02-24 09:22:16,546 [I] Running command `service evmserverd status` (utils/ssh.py:138)
2016-02-24 09:22:16,876 [E] Error setting up provider rhos7-ga (fixtures/provider.py:43)

[....]

2016-02-24 09:22:22,868 [I] ------------------ test_ssa_packages[rhos7-ga-fedora-22-ext4] ------------------ (cfme/tests/infrastructure/test_instance_analysis.py:438)
2016-02-24 09:22:22,894 [E] @pytest.fixture(scope='module')
    def setup_provider_modscope(request, provider):
        """Function-scoped fixture to set up a provider"""
>       _setup_provider(provider.key, request)

fixtures/provider.py:58: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
fixtures/provider.py:46: in _setup_provider
    skip(provider_key)
fixtures/provider.py:25: in skip
    name, location = get_test_idents(node)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def get_test_idents(item):
>       return item.location[2], item.location[0]
E       AttributeError: 'Module' object has no attribute 'location'

fixtures/artifactor_plugin.py:45: AttributeError (fixtures/artifactor_plugin.py:45)
```

**Comment by psav** [comment]
@RonnyPfannschmidt can you take a look at this please?

File: fixtures/artifactor_plugin.py [hunk 1]
```diff
@@ -42,7 +42,10 @@
 
 
 def get_test_idents(item):
-    return item.location[2], item.location[0]
+    try:
+        return item.location[2], item.location[0]
+    except Exception:
+        return (None, None)
```
Comment (RonnyPfannschmidt) [inline fixtures/artifactor_plugin.py]: we should try to include item.fspath if the attribute exists

Comment (vrutkovs) [inline fixtures/artifactor_plugin.py]: Okay, although its not easy to reproduce this one

**Comment by dajoRH** [comment]
Lint report for commit 06f1f6194c3e274324a876bce143ff9d2e231f86:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by RonnyPfannschmidt** [comment]
looks good to me now :)

**Comment by mfalesni** [comment]
OK, merging then.

**Pull request merged**